src/script/lua_api/l_client.cpp

```diff
@@ -251,6 +252,50 @@ int ModApiClient::l_get_server_info(lua_State *L)
 	return 1;
 }
 
+// get_item_def(itemstring)
+int ModApiClient::l_get_item_def(lua_State *L)
+{
+	IGameDef *gdef = getGameDef(L);
+	assert(gdef);
+
+	IItemDefManager *idef = gdef->idef();
+	assert(idef);
+	
+	if (!lua_isstring(L, 1))
+		return 0;
+
+	const std::string &name = lua_tostring(L, 1);
```

Comment:
i don't think this works on every compiler as lua_tostring provides a const char * and you have a const ref std::string, here a copy is required